File: TP01/TP01Q01 - Palíndromo em Java/Palindromo.java
class Palindromo {

	public static boolean isPalindromo(String word) { // Método que verifica se uma palavra é palíndroma
		boolean resposta = true; // Por padrão, a palavra é palíndroma até que se encontre um caracter que não
									// seja igual ao seu respectivo caracter na posição oposta

		for (int i = 0; i < word.length(); i++) { // Para cada caracter da palavra
			if (word.charAt(i) != word.charAt(word.length() - 1 - i)) { // Caso seja diferente do caracter da posição
																		// oposta
				resposta = false;
			}
		}

		return resposta;
	}

	public static boolean isFim(String s) {
		return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
	}

	public static void main(String[] args) {
		String[] entrada = new String[1000];
		int i = 0; // Index da entrada

		// Leitura da entrada padrao
		do {
			entrada[i] = MyIO.readLine();
		} while (isFim(entrada[i++]) == false);
		i--; // Desconsiderar ultima linha contendo a palavra FIM

		for (int j = 0; j < i; j++) {
			// Para cada linha da entrada, verifica se é palíndroma
			if (isPalindromo(entrada[j])) {
				MyIO.println("SIM");
			} else {
				MyIO.println("NAO");
			}
		}
	}
}

File: TP01/TP01Q02 - Palíndromo em C/Palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool isPalindromo(char word[])
{
    // Método que verifica se uma palavra é palíndroma
    bool resposta = true; // Por padrão, a palavra é palíndroma até que se encontre um caracter que não
                          // seja igual ao seu respectivo caracter na posição oposta
    for (int i = 0; i < strlen(word); i++)
    { // Para cada caracter da palavra
        if (word[i] != word[strlen(word) - 1 - i])
        { // Caso seja diferente do caracter da posição
          // oposta
            resposta = false;
        }
    }

    return resposta;
}

int main(void)
{
    char entrada[1000];

    while (strcmp(entrada, "FIM") != 0) // Enquanto a entrada não for FIM
    {
        scanf(" %[^\n]s", entrada);      // Ler a proxima string
        if (strcmp(entrada, "FIM") != 0) // verifica se é FIM
        {
            // Caso seja palíndroma, imprime SIM, caso contrário, imprime NAO
            if (isPalindromo(entrada))
            {
                printf("SIM\n");
            }
            else
            {
                printf("NAO\n");
            }
        }
    }

    return 0;
}
